scripts/corelli/calibration/utils.py

```diff
@@ -56,4 +125,34 @@ def calculate_tube_calibration(workspace: InputWorkspace, tube_name: str, shadow
     wire_pixel_positions = (PIXELS_PER_TUBE / TUBE_LENGTH) * wire_positions + PIXELS_PER_TUBE / 2
     fit_par = TubeCalibFitParams(wire_pixel_positions, height=peak_height, width=peak_width, margin=fit_domain)
     fit_par.setAutomatic(True)
-    return tube.calibrate(workspace, tube_name, wire_positions, peaks_form, fitPar=fit_par)
+    calibration_table, _ = tube.calibrate(workspace, tube_name, wire_positions, peaks_form, fitPar=fit_par,
+                                          outputPeak=True)
+    return calibration_table
+
+
+def apply_calibration(workspace: WorkspaceTypes, calibration_table: InputTable,
+                      output_workspace: Optional[str] = None, show_instrument: bool = False) -> Workspace2D:
+    r"""
+    Calibrate the detector positions with an input table, and open the instrument view if so requested.
+
+    :param workspace: input Workspace2D containing total neutron counts per pixel
+    :param calibration_table: a TableWorskpace containing one column for detector ID and one column
+    for its calibrated XYZ coordinates, in meters
+    :param output_workspace: name of the output workspace containing calibrated detectors
+    :param show_instrument: open the instrument view for `output_workspace`
+
+    :raises AssertionError: either `workspace` or `calibration_table` are not found
+    """
+    assert AnalysisDataService.doesExist(str(workspace)), f'No worksapce {str(workspace)} found'
+    assert AnalysisDataService.doesExist(str(calibration_table)), f'No table {str(calibration_table)} found'
+    if output_workspace is None:
+        output_workspace = str(workspace) + '_calibrated'
+
+    CloneWorkspace(InputWorkspace=workspace, OutputWorkspace=output_workspace)
+    ApplyCalibration(Workspace=output_workspace, CalibrationTable=calibration_table)
+
+    if show_instrument is True and None not in (InstrumentViewPresenter, InstrumentViewPresenter):
+        instrument_presenter = QAppThreadCall(InstrumentViewPresenter)(mtd[output_workspace])
```

Comment:
I didn't realize this function existed. Nice!

Reply:
neither did I until Andrei asked a related questions two days ago in Mantid slack :smile: